homeassistant/components/xiaomi_ble/binary_sensor.py

```diff
@@ -28,20 +28,48 @@
 from .device import device_key_to_bluetooth_entity_key
 
 BINARY_SENSOR_DESCRIPTIONS = {
-    XiaomiBinarySensorDeviceClass.MOTION: BinarySensorEntityDescription(
-        key=XiaomiBinarySensorDeviceClass.MOTION,
-        device_class=BinarySensorDeviceClass.MOTION,
+    XiaomiBinarySensorDeviceClass.DOOR: BinarySensorEntityDescription(
+        key=XiaomiBinarySensorDeviceClass.DOOR,
+        device_class=BinarySensorDeviceClass.DOOR,
     ),
     XiaomiBinarySensorDeviceClass.LIGHT: BinarySensorEntityDescription(
         key=XiaomiBinarySensorDeviceClass.LIGHT,
         device_class=BinarySensorDeviceClass.LIGHT,
     ),
+    XiaomiBinarySensorDeviceClass.MOISTURE: BinarySensorEntityDescription(
+        key=XiaomiBinarySensorDeviceClass.MOISTURE,
+        device_class=BinarySensorDeviceClass.MOISTURE,
+    ),
+    XiaomiBinarySensorDeviceClass.MOTION: BinarySensorEntityDescription(
+        key=XiaomiBinarySensorDeviceClass.MOTION,
+        device_class=BinarySensorDeviceClass.MOTION,
+    ),
+    XiaomiBinarySensorDeviceClass.OPENING: BinarySensorEntityDescription(
+        key=XiaomiBinarySensorDeviceClass.OPENING,
+        device_class=BinarySensorDeviceClass.OPENING,
+    ),
     XiaomiBinarySensorDeviceClass.SMOKE: BinarySensorEntityDescription(
         key=XiaomiBinarySensorDeviceClass.SMOKE,
         device_class=BinarySensorDeviceClass.SMOKE,
     ),
-    XiaomiBinarySensorDeviceClass.MOISTURE: BinarySensorEntityDescription(
-        key=XiaomiBinarySensorDeviceClass.MOISTURE,
+    "device_forcibly_removed": BinarySensorEntityDescription(
+        key="device_forcibly_removed",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    "door_left_open": BinarySensorEntityDescription(
+        key="door_left_open",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    "door_stuck": BinarySensorEntityDescription(
+        key="door_stuck",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    "knock_on_the_door": BinarySensorEntityDescription(
+        key="knock_on_the_door",
+    ),
+    "pry_the_door": BinarySensorEntityDescription(
+        key="pry_the_door",
```

Comment:
Should these keys be moved to the const file for clarity of transfer into the library where the other SensorDeviceClasses come from, whenever that happens?

Reply:
@bdraco what do you think? Should I move these to sensor-state-data, leave it here, or move it to const.py?

Reply:
If they are going to be used in other integrations `sensor-state-data` makes sense. If its only here than its probably best to leave as-is

Reply:
These binary sensors are very specific. Don’t think they will be reused somewhere else soon. Let’s keep it here.

Reply:
It would still be nice to extend the enum in `xiaomi-ble` so we don't have bare strings here.

Reply:
Ok, will do that tomorrow.

Reply:
@bdraco If I understand it correct, it isn't allowed to extend a non-empty Enum. I tried that here, but it won't pass the tests.

https://github.com/Bluetooth-Devices/xiaomi-ble/pull/29

All I can do is create a second Enum with the Xiaomi specific sensors, I guess.

Reply:
To extend an enum you have to make a copy of it manually. The alternate solution is fine as well.